### PR TITLE
Restructure documentation

### LINKED ISSUE #429: Revise documentation structure
The structure and major topics of the documentation should be revised.

### PR DESCRIPTION
**Pull request by sondreso**
**Issue**
Resolves #429 

**Approach**
_Short description of the approach_

URL to test: file:///scratch/ert/sonso/docs/index.html

### CODE REVIEW

**Comment by EivindBakken** [comment]
General proposals for several of the introduction chapters: Note that there are many users of ERT that doesn't do model updating (history matching), eg. geologist and reservoir engineers using ERT as a work flow manager for FMU on field development projects (no history yet). So for instance the start page manual.html should reflect that a) ERT is a work flow manager that b) can be used for assisted history match/model update. We should discuss/align wording before implementation (and I need to be sure whether this is primarily a Equinor manual)

**Comment by EivindBakken** [comment]
The GENDATA_RFT functionality seems to be missing in the documentation

**Comment by mortalisk** [comment]
Here is an initial suggestion for structure, which I have started implementing. Please edit and add stuff if you have ideas.

### Welcome
 - Welcome text (link to theroy)

### Getting Started
A chapter for beginners
- **Installation**
	- Explain how to install (link to github?)
	- Explain how to source komodo?

- **Configuration guide**
	- How to create an entire config directory (recreate poly_eval?)
	- Several pages with link to relevant chapters

 - **Launching**
	- need ert installation (link Installation and  configuration (link to Configuration Guide)
	- navigate to config folder
	- run "ert gui config.ert" (mention CLI, link to Running ERT)
	- (link to gui guide)

- **Gui guide (with screenshots?)**
	- Repeat how to launch
	- Explain the gui

### Theory
- Enamble based methods

### Reference
Place to look up details
- Running Ert
- Configuration reference
	- Data Types
	- etc...
- Workflows
	- External Workflows
	- Etc.
- Advanced

### About
Info about this release, etc..

File: docs/rst/manual/reference/workflows/configuring_jobs.rst [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+keyword you can request the loading of an external shared library:
+
+::
+
+    INTERNAL  TRUE                     -- The job will call an internal function of the current running ERT instance.
+    FUNCTION  enkf_main_plot_all       -- Name of the ERT function we are calling; must be marked exportable.
+    MODULE    /name/of/shared/library  -- Very optional - to load an extra shared library.
+
+
+Configure a an internal job: Python
```
Comment (oyvindeide) [inline docs/rst/manual/reference/workflows/configuring_jobs.rst]: An extra: `a`

File: docs/rst/manual/getting_started/setup.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    source /project/res/komodo/stable/enable
+
+Then you can test the installation by running::
+
+    ert --help
+
+If you get a helful message explaining how to use ERT, you are ready.
+
+If you don't have access to komodo, you might have another means of using it in your
+organisation. You should ask you colleagues.
```
[the remark this answers is not in the document]
Comment (markusdregi) [reply docs/rst/manual/getting_started/setup.rst]: I think we should leave a single non-general comment in the `getting_started` section regarding setup. Besides this, we should indeed be tool specific.

File: docs/rst/manual/reference/workflows/external.rst [hunk 3]
```diff
@@ -0,0 +1,12 @@
+External Workflows
+==================
+
+These jobs invoke an external program/script to do the job, this is
+very similar to the jobs of the forward model, but instead of running
+as separate jobs on the cluster - one for each realization, the
+workflow jobs will be invoked on the workstation running ert, and
```
Comment (oyvindeide) [inline docs/rst/manual/reference/workflows/external.rst]: Think we're using `ERT` in `ALL CAPS`?

File: docs/rst/manual/reference/workflows/internal.rst [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+core code and a new version must be installed.
+
+.. _ert_script:
+
+Run a Python Script
+-------------------
+
+If you are using one of the Python based frontends, *gert* or
+*erthsell*, you can write your own Python script which is run as part
+of the existing process. By using the full ert Python api you get
```
Comment (oyvindeide) [inline docs/rst/manual/reference/workflows/internal.rst]: `ERT`

File: docs/rst/manual/reference/workflows/internal.rst [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+access to powerful customization/automization features. Below is an
```
Comment (oyvindeide) [inline docs/rst/manual/reference/workflows/internal.rst]: This sentence might be an understatement ;) Also, `automization` is misspelled, should be `automation`

**Comment by oyvindeide** [comment]
One thing I've noticed is that when I press an item in the navigation bar, it jumps out one indent, looking like it belongs to the item above.

File: docs/rst/manual/reference/workflows/internal.rst (hunk 4 above)
Comment (oyvindeide) [reply docs/rst/manual/reference/workflows/internal.rst]: Forgot ERT

**Pull request merged**